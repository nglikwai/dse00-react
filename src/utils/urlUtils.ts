import { IStringifyOptions, stringify } from 'qs'

export const buildQuery = (
  path: string,
  queryObject: Record<string, string | string[]>,
  options?: IStringifyOptions
): string =>
  path +
  stringify(queryObject, {
    encode: false,
    arrayFormat: 'comma',
    addQueryPrefix: true,
    ...options,
  })
