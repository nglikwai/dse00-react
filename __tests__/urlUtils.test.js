import { buildQuery } from 'src/utils/urlUtils'

describe('utils/urlUtils:buildQuery >', () => {
  const link = 'https://{hostname}/article/{articleKey}'

  it('should return url with queries on proper input', () => {
    const result = buildQuery(link, {
      utm_source: 'share_button',
      utm_medium: 'referral',
      utm_campaign: 'share_web',
    })

    expect(result).toMatchSnapshot()
  })

  it('should return original url on no query input', () => {
    const result = buildQuery(link)

    expect(result).toMatchSnapshot()
  })
})
