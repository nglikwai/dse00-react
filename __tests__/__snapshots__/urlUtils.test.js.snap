// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/urlUtils:buildQuery > should return original url on no query input 1`] = `"https://{hostname}/article/{articleKey}"`;

exports[`utils/urlUtils:buildQuery > should return url with queries on proper input 1`] = `"https://{hostname}/article/{articleKey}?utm_source=share_button&utm_medium=referral&utm_campaign=share_web"`;
